app/assets/javascripts/discourse/lib/plugin-api.js.es6

```diff
@@ -334,6 +335,22 @@ class PluginApi {
   }
 
   /**
+   * Adds a panel to the header
+   *
+   * takes a widget name, a value to toggle on, and a function which returns the attrs for the widget
+   * Example:
+   * ```javascript
+   * api.addHeaderPanel('widget-name', 'widgetVisible', function(attrs, state) { return {}; });
+   * ```
+   * note that 'toggle' is an attribute on the state of the header widget,
+   * and the attrFn receives the current attrs and state of the header as arguments,
+   * and returns a hash of attrs to pass to the widget
+   **/
+   addHeaderPanel(name, toggle, attrFn) {
+     queryRegistry('header').prototype.attachAdditionalPanel(name, toggle, attrFn);
```

Comment:
Rather than messing with the registry and prototypes, why not make `additionalPanels` a private local variable in the `header.js.es6` file?

Then export the function `attachAdditionalPanel` and just call it in here. That is much more in line with how we usually do things.

Also, `attrFn` is not super clear to me. It seems to me like it's more of a transformer. Maybe call it `transformAttrs` instead, and update the example so it shows why you might want to do that?